src/tetris.py

```diff
@@ -19,37 +17,89 @@
 # Define the dimensions of screen object and caption
 screen = pygame.display.set_mode((grid.width, grid.height)) 
 pygame.display.set_caption("Tetris Demo")
+clock = pygame.time.Clock()
 
 # Variable to keep our game loop running 
 gameOn = True
 
 # Set screen black initially
 screen.fill((0,0,0))
 
-
+'''
 # Sets each block in the grid to a random color
 for x in grid.grid:
 	for y in x:
 		y.surf.fill( ( random.randint(0,255),random.randint(0,255),random.randint(0,255) ) )  
 		screen.blit(y.surf, y.position)
+'''
 
-# Our game loop 
-while gameOn: 
-	# for loop through the event queue 
-	for event in pygame.event.get(): 
-		
-		# Check for KEYDOWN event 
-		if event.type == KEYDOWN: 
+pos = [4,0]
+
+blk = grid.grid[pos[0]][pos[1]]
+
+blk.color = (0,200,255)
+blk.surf.fill( (0,200,255) )  
+screen.blit( blk.surf, blk.position )
+
+drop_speed_control = timer.Timer()
+drop_speed = 1000
+
+try:
+	# Our game loop 
+	while gameOn: 
+		# for loop through the event queue 
+		for event in pygame.event.get(): 
 			
-			# If the Backspace key has been pressed set 
-			# running to false to exit the main loop 
-			if event.key == K_BACKSPACE: 
+			# Check for KEYDOWN event 
+			if event.type == KEYDOWN: 
+				
+				# If the Backspace key has been pressed set 
+				# running to false to exit the main loop 
+				if event.key == K_BACKSPACE: 
+					gameOn = False
+
+				if event.key == K_LEFT:
+					pos = grid.move_left(screen, pos)
+					pygame.display.flip()
+
+				elif event.key == K_RIGHT:
+					pos = grid.move_right(screen, pos)
+					pygame.display.flip()
+
+				elif event.key == K_DOWN:
+					drop_speed -= 100
```

Comment:
later we will want to make sure the drop speed gets reset back to normal when a new shape comes down.

Reply:
so the lower `drop_speed` is, the faster it drops? confusing variable name. maybe `drop_delay`